### ISSUE TITLE
Unable to replicate experiment results

### ISSUE THREAD

**Issue by felixhao28** [kimiyoung/transformer-xl#14]
I am not sure where it is wrong but I have been training enwik8 with your TensorFlow code and default parameters on 4 GPUs for 4 days and the loss never drops below 4.2 meanwhile the learning rate already drops to 0.000001. Is there any special tricks to replicate the experiment?

Thanks.

P.S. I am using Python 3 and TensorFlow 1.11.0. I have not tried on the other 3 datasets yet. I also tried transformer-xl on a private dataset (where a single-layer word-level LSTM can achieve around 60%+ accuracy), and its loss also never drops below 4.2 and accuracy never goes higher than 15%.

**Comment by felixhao28** [comment]
> Is it possible for you to share your commands?

I simply followed the instructions. The commands were:

```bash
/transformer-xl$ bash getdata.sh
/transformer-xl/tf$ bash scripts/enwik8_gpu.sh train_data
/transformer-xl/tf$ bash scripts/enwik8_gpu.sh train
```

I made some slight changes to the code to add accuracy metrics. I will upload the code today.

**Comment by felixhao28** [comment]
FYI, here's the md5sum for data/enwiki8/train.txt:

5b19721ed0e11b5edb487ed29c59dc11  data/enwik8/train.txt

**Comment by felixhao28** [comment]
The code is uploaded. Here's the change: https://github.com/kimiyoung/transformer-xl/compare/master...felixhao28:master

**Comment by felixhao28** [comment]
No worries. It is most likely something wrong on my side rather than yours. I simply don't know how to figure it out by myself. Thanks you for your great support!

**Comment by kimiyoung** [comment]
It turns out that there has been an error in the scripts we provided. Sorry for the inconvenience. We have updated the scripts and the README to fix the issue.

Quoting from the new README:

> Note that our state-of-the-art results reported in the paper were obtained by training the model on a large-scale TPU cluster, and our gpu codebase currently does not support distributed training. Here we provide two sets of hyperparameters and scripts:
> 
> *large_tpu.sh are for the SoTA setting on TPUs. These are exactly the commands we used to obtained our best results.
> *base_gpu.sh are for the base models which can be run on a few GPUs.

The reason of the problem you reported is that we used a large model setting in the previous script, but forgot to update the `warmup_steps` parameter, and as a result the large model was not correctly optimized.

In the new version, all GPU scripts are for base models, but it is possible to increase the model sizes so that the model size gets closer to the SoTA TPU setting. Quoting again from the new README:

> If enough resources are available, increasing the model sizes (e.g., N_LAYER, D_MODEL, D_EMBED, D_HEAD, D_INNER) so that they are closer to the values defined in scripts/dataset_large_tpu.sh. Likewise, when resources are limited, decrease the model sizes. It is recommended to ensure that D_MODEL == D_EMBED and D_MODEL == N_HEAD x D_HEAD. When the model sizes increase, remember to increase warmup_steps accordingly to alleviate optimization difficulties.

Hope this helps. Let us know if this works for you.

**Comment by felixhao28** [comment]
I will try it today. Thanks.

**Comment by felixhao28** [comment]
What will be the expected loss/accuracy for the gpu base model?

```
# Model
N_LAYER=12
D_MODEL=512
D_EMBED=512
N_HEAD=8
D_HEAD=64
D_INNER=2048
```

**Comment by kimiyoung** [comment]
bpc ~1.06 for enwik8.
